fletcher/string_array.py

```diff
@@ -269,8 +269,18 @@ def _call_str_accessor(self, func, *args, **kwargs) -> pd.Series:
         """Call the str accessor function with transforming the Arrow series to pandas series
          and back."""
         pd_series = self.data.to_pandas()
-        array = pa.array(getattr(pd_series.str, func)(*args, **kwargs).values)
-        return self._series_like(array)
+        pd_result = getattr(pd_series.str, func)(*args, **kwargs)
+        if isinstance(pd_result, pd.DataFrame):
+            for c in pd_result.columns:
+                pd_result[c] = type(self.obj.values)(
+                    pd_result[c].replace({np.nan: None}).values
```

Comment:
Why aren't you using `self.series_like` here?

Reply:
`_series_like` is not taking care of the index yielded by `str.extractall` and similar functions.